caffe2/python/tutorials/MNIST.ipynb

```diff
@@ -553,6 +553,10 @@
     "workspace.RunNetOnce(train_model.param_init_net)\n",
     "# creating the network\n",
     "workspace.CreateNet(train_model.net, overwrite=True)\n",
+    "# Also initialize the test model before training\n",
+    "workspace.RunNetOnce(test_model.param_init_net)\n",
```

Comment:
Also init_param=False on the model should take care of not overriding them. Could you check what is in param_init_net to see if it overrides something? 

See conv.py and fc.py, there we don't initialize parameters if model.init_params == False